test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -109,6 +110,146 @@ TEST(QuicPlatformTest, QuicPtrUtil) {
   EXPECT_EQ("aaa", *p);
 }
 
+namespace {
+class QuicLogThresholdSaver {
+public:
+  QuicLogThresholdSaver()
+      : level_(quic::GetLogger().level()), verbosity_threshold_(quic::GetVerbosityLogThreshold()) {}
+
+  ~QuicLogThresholdSaver() {
+    quic::SetVerbosityLogThreshold(verbosity_threshold_);
+    quic::GetLogger().set_level(level_);
+  }
+
+private:
+  const quic::QuicLogLevel level_;
+  const int verbosity_threshold_;
+};
+} // namespace
+
+TEST(QuicPlatformTest, QuicLog) {
+  QuicLogThresholdSaver saver;
+
+  // By default, tests emit logs at level ERROR or higher.
+  ASSERT_EQ(quic::ERROR, quic::GetLogger().level());
+
+  int i = 0;
+
+  QUIC_LOG(INFO) << (i = 10);
```

Comment:
Mind using EXPECT_LOG_[NOT_]ONTAINS macros instead of or in additon to the math checks?
It'll also verify we did the log mapping correctly.

Reply:
I added some EXPECT_LOG_CONTAINS.
These macros forcefully set the log level to trace([example](https://github.com/envoyproxy/envoy/blob/master/test/test_common/logging.h#L157)), so things like EXPECT_NO_LOGS do not work for the test where log is not emitted due to log level.

After this PR, I'll work on quic_mock_log_impl.h and add tests for the "no log expected" case.